js/ad_manager_vast.js

```diff
@@ -1494,6 +1494,12 @@ OO.Ads.manager(function(_, $) {
             _skipAd(currentAd);
           }
         }
+        // [PLAYER-3912]
+        // Make sure that NONLINEAR_AD_PLAYED gets fired when a stream ends
+        // with an active overlay, otherwise the skin will keep it for new videos.
+        else if (!ad.isLinear && params.code === this.amc.AD_CANCEL_CODE.STREAM_ENDED) {
+          _endAd(ad, false);
```

Comment:
Not sure whether I should use `_skipAd` or `_endAd` here, but I think `_endAd` makes more sense since skipping is usually a manual action and this is sort of a side-effect of switching videos.

Reply:
I think endAd is the better thing to use. Just in case we fire off something on the message bus we don't want to report another skip for analytics.